legion/legion/k8s/services.py

```diff
@@ -0,0 +1,474 @@
+#
+#    Copyright 2018 EPAM Systems
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+"""
+legion k8s services classes
+"""
+import logging
+
+import kubernetes
+import kubernetes.client
+import kubernetes.config
+import kubernetes.config.config_exception
+
+import legion.model
+from legion.containers.headers import DOMAIN_MODEL_ID, DOMAIN_MODEL_VERSION
+import legion.k8s.enclave
+from legion.k8s.definitions import ModelIdVersion
+from legion.k8s.definitions import LEGION_COMPONENT_LABEL, LEGION_SYSTEM_LABEL, LEGION_API_SERVICE_PORT
+from legion.k8s.definitions import STATUS_OK, STATUS_WARN, STATUS_FAIL
+import legion.k8s.utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Service:
+    """
+    Data-structure for describing K8S service
+    """
+
+    def __init__(self, k8s_service):
+        """
+        Build service structure from K8S Service
+
+        :param k8s_service: K8S service
+        :type k8s_service: V1Service
+        """
+        if not k8s_service.metadata.labels:
+            raise Exception('Cannot construct Service from K8S service without any labels')
+
+        self._name = k8s_service.metadata.labels.get(LEGION_COMPONENT_LABEL)
+
+        if not self._name:
+            raise Exception('Cannot construct Service from K8S service without label {}'.format(LEGION_COMPONENT_LABEL))
```

Comment:
Non-informative

Reply:
Why? Code says that you are trying to construct `Service` instance from wrong k8s service object

Reply:
"Invalid service for introspection: label {} missed"